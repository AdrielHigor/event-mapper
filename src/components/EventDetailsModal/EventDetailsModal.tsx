import React from "react";
import "./EventDetailsModal.css";
import ModalContainer from "../ModalContainer/ModalContainer";
import { IEventResponse } from "../../utils/interfaces/base";

interface IEventDetailsModal {
  modalIsOpen: boolean;
  closeModal: (
    event: React.MouseEvent<Element, MouseEvent> | React.KeyboardEvent<Element>
  ) => void;
  customClassName?: string;
  event?: IEventResponse;
  onDelete: (id: number) => void;
}

const EventDetailsModal = ({
  modalIsOpen,
  closeModal,
  customClassName,
  event,
  onDelete,
}: IEventDetailsModal) => {
  const formatDateTime = (date: string) => {
    console.log(date);
    const dateObj = new Date(date);

    return `${dateObj.getDate().toString().padStart(2, "0")}/${(
      dateObj.getMonth() + 1
    )
      .toString()
      .padStart(2, "0")}/${dateObj.getFullYear()} Ã s ${dateObj
      .getUTCHours()
      .toString()
      .padStart(2, "0")}:${dateObj.getMinutes().toString().padStart(2, "0")}`;
  };

  return (
    <ModalContainer
      modalIsOpen={modalIsOpen}
      closeModal={closeModal}
      customClassName={customClassName}
    >
      <div className="first-steps">
        <div className="modal-header">
          <h2>{event?.name}</h2>
          <p className="description">{event?.description}</p>
        </div>
        <div>
          {event?.startDateTime && (
            <div>
              <p className="sub-title">Data e Hora Inicial</p>
              <p>{formatDateTime(event.startDateTime)}</p>
            </div>
          )}
          {event?.endDateTime && (
            <div>
              <p className="sub-title">Data e Hora Final</p>
              <p>{formatDateTime(event.endDateTime)}</p>
            </div>
          )}
        </div>
        <div className="options">
          <button className="confirm-button" onClick={closeModal}>
            Concluir!
          </button>
          <button
            className="cancel-button"
            onClick={(e) => (event ? onDelete(event.id) : closeModal(e))}
          >
            Deletar
          </button>
        </div>
      </div>
    </ModalContainer>
  );
};

export default EventDetailsModal;
