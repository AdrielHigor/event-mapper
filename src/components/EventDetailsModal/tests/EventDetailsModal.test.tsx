import renderer from "react-test-renderer";
import EventDetailsModal, { IEventDetailsModal } from "../EventDetailsModal";

jest.mock("../EventDetailsModal.css", () => ({}));

describe("EventDetailsModal", () => {
  let component: renderer.ReactTestRenderer;

  const defaultProps: IEventDetailsModal = {
    modalIsOpen: true,
    closeModal: jest.fn(),
    customClassName: "test-class",
    event: {
      id: 1,
      name: "Test event",
      description: "Test description",
      startDateTime: "2023-05-01T12:00:00.000Z",
      endDateTime: "2023-05-01T14:00:00.000Z",
      createdAt: "2023-05-01T12:00:00.000Z",
      deletedAt: "2023-05-01T12:00:00.000Z",
      updatedAt: "2023-05-01T12:00:00.000Z",
      location: {
        id: 1,
        point: {
          coordinates: [0, 1],
          crs: {
            properties: {
              type: "test",
            },
            type: "test",
          },
          type: "test",
        },
      },
    },
    onDelete: jest.fn(),
  };

  beforeEach(() => {
    component = renderer.create(<EventDetailsModal {...defaultProps} />);
  });

  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render event details", () => {
    const title = component.root.findByProps({
      className: "modal-header-title",
    });
    const description = component.root.findByProps({
      className: "description",
    });
    const subTitles = component.root.findAllByProps({
      className: "sub-title",
    });
    const startDateTimeElement = subTitles[0].children[0];
    const endDateTimeElement = subTitles[1].children[0];

    expect(title.props.children).toBe(defaultProps.event.name);
    expect(description.props.children).toBe(defaultProps.event.description);
    expect(startDateTimeElement).toBe("Data e Hora Inicial");
    expect(endDateTimeElement).toBe("Data e Hora Final");
  });

  it("should call onDelete when delete button is clicked", () => {
    const deleteButton = component.root.findByProps({
      className: "cancel-button",
    });
    deleteButton.props.onClick();

    expect(defaultProps.onDelete).toHaveBeenCalledTimes(1);
    expect(defaultProps.onDelete).toHaveBeenCalledWith(defaultProps.event.id);
  });

  it("should call closeModal when confirm button is clicked", () => {
    const confirmButton = component.root.findByProps({
      className: "confirm-button",
    });
    confirmButton.props.onClick();

    expect(defaultProps.closeModal).toHaveBeenCalledTimes(1);
  });
});
