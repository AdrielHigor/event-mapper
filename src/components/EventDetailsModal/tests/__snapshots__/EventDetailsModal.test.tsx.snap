// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetailsModal should render correctly 1`] = `
<div>
  <div
    className="first-steps"
  >
    <div
      className="modal-header"
    >
      <h2
        className="modal-header-title"
      >
        Test event
      </h2>
      <p
        className="description"
      >
        Test description
      </p>
    </div>
    <div>
      <div>
        <p
          className="sub-title"
        >
          Data e Hora Inicial
        </p>
        <p>
          01/05/2023 às 12:00
        </p>
      </div>
      <div>
        <p
          className="sub-title"
        >
          Data e Hora Final
        </p>
        <p>
          01/05/2023 às 14:00
        </p>
      </div>
    </div>
    <div
      className="options"
    >
      <button
        className="confirm-button"
        onClick={[MockFunction]}
      >
        Concluir!
      </button>
      <button
        className="cancel-button"
        onClick={[Function]}
      >
        Deletar
      </button>
    </div>
  </div>
</div>
`;
